agent/engine/execcmd/agent_version_test.go

```diff
@@ -15,108 +15,205 @@ package execcmd
 
 import (
 	"errors"
+	"io/ioutil"
+	"os"
+	"sort"
 	"testing"
+	"time"
 
 	"github.com/stretchr/testify/assert"
 )
 
-func TestCompareAgentVersion(t *testing.T) {
+func TestAgentVersionCompare(t *testing.T) {
 	tt := []struct {
-		name        string
-		v1, v2      string
-		res         int
-		errExpected bool
+		name     string
+		lhs, rhs agentVersion
+		res      int
 	}{
 		{
-			"v1 < v2", "3.0.236.0", "3.0.237.0", -1, false,
+			"lhs < rhs", []uint{3, 0, 236, 0}, []uint{3, 0, 237, 0}, -1,
 		},
 		{
-			"v1 == v2", "3.0.236.0", "3.0.236.0", 0, false,
+			"lhs == rhs", []uint{3, 0, 236, 0}, []uint{3, 0, 236, 0}, 0,
 		},
 		{
-			"v1 > v2", "3.0.237.0", "3.0.236.0", 1, false,
+			"lhs > rhs", []uint{3, 0, 237, 0}, []uint{3, 0, 236, 0}, 1,
 		},
 		{
-			"negative v1", "-1.0.0.0", "3.0.236.0", badInputReturn, true,
+			"fewer elements lhs", []uint{3, 0, 236}, []uint{1, 1, 1, 1}, 1,
 		},
 		{
-			"negative v2", "3.0.236.0", "-1.0.0.0", badInputReturn, true,
+			"fewer elements rhs", []uint{1, 1, 1, 1}, []uint{3, 0, 236}, -1,
 		},
 		{
-			"invalid v1", "a.b.c.d", "3.0.236.0", badInputReturn, true,
+			"blank lhs", []uint{}, []uint{1, 1, 1, 1}, -1,
 		},
 		{
-			"invalid v2", "3.0.236.0", "a.b.c.d", badInputReturn, true,
-		},
-		{
-			"missing elements v1", "3.0.236", "1.1.1.1", badInputReturn, true,
-		},
-		{
-			"missing elements v2", "1.1.1.1", "3.0.236", badInputReturn, true,
-		},
-		{
-			"blank v1", "", "1.1.1.1", badInputReturn, true,
-		},
-		{
-			"blank v2", "1.1.1.1", "", badInputReturn, true,
+			"blank rhs", []uint{1, 1, 1, 1}, []uint{}, 1,
```

Comment:
blank both sides returns 0?

Reply:
It should, let me add that case here though.

Reply:
added test case [here](https://github.com/aws/amazon-ecs-agent/pull/2701/files#diff-6cf239de592683f4c177ec789f5615ded75e41e50ea468fbfc8531930714efffR37)